src/batch/src/executor/update.rs

```diff
@@ -165,24 +167,19 @@ impl UpdateExecutor {
                     unreachable!("no chunk should be yielded when appending the deleted row as the chunk size is always even");
                 };
                 if let Some(chunk) = builder.append_one_row(row_insert) {
-                    notifiers.push(write_txn_data(chunk).await?);
+                    rows_updated += write_txn_data(chunk).await?;
                 }
             }
         }
 
         if let Some(chunk) = builder.consume_all() {
-            notifiers.push(write_txn_data(chunk).await?);
+            rows_updated += write_txn_data(chunk).await?;
         }
 
-        notifiers.push(write_handle.end()?);
-
-        // Wait for all chunks to be taken / written.
-        let rows_updated = try_join_all(notifiers)
+        write_handle
+            .end()?
```

Comment:
I guess we can further make `end` an async function by `await`ing the notifier inside its implementation, so that callers will not ignore this step.